### PR TITLE
ADDED new messages to FB connector

### PR DESCRIPTION
**Pull request by advl03**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [X] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by advl03** [comment]
Did you take a look at the PR? Is there anything I need to change/add?

**Comment by amn41** [comment]
thanks @advl03 for creating this PR! Looks super useful, sorry we seem to have dropped this. We will take a look and someone will review.

File: rasa_core/dispatcher.py [hunk 1]
```diff
@@ -71,6 +71,23 @@ def utter_message(self, text):
...
+                                        ))
+        self.output_channel.send_quick_replies(self.sender_id,
+                                               text,
+                                               quick_replies)
+
+    def utter_send_template(self, attachment):
+        self.latest_bot_messages.append(BotMessage(text=None,
+                                        data={"attachment": attachment}
+                                        ))
+        self.output_channel.send_template(self.sender_id, attachment)
```
Comment (akelad) [inline rasa_core/dispatcher.py]: what about the send_list/send_generic_templates?

Comment (advl03) [reply rasa_core/dispatcher.py]: It seems they are not in the dispatcher code. As I've got a deadline on Monday, I have no time to add the proper lines:

def utter_send_list_template(self, attachment):
        self.latest_bot_messages.append(BotMessage(text=None, data={"attachment": attachment}))
        self.output_channel.send_list_template(self.sender_id, attachment)

    def utter_send_generic_template(self, attachment, meta= None, square=False):
        self.latest_bot_messages.append(BotMessage(text=None, data={"attachment": attachment}))
        self.output_channel.send_generic_template(self.sender_id, attachment, meta,square)

Comment (akelad) [reply rasa_core/dispatcher.py]: that's fine, it's not urgent, feel free to make the changes whenever you have time

Comment (akelad) [reply rasa_core/dispatcher.py]: @advl03 any chance of updating this?

Comment (advl03) [reply rasa_core/dispatcher.py]: Updated both dispatcher and changelog. Sorry for the delay!

**Comment by akelad** [comment]
Looks great, thanks a lot for submitting this! Travis is failling on some pep8 checks now though, would be great if you could fix those https://travis-ci.com/RasaHQ/rasa_core/jobs/155499104#L1257

**Comment by akelad** [comment]
Actually hold on, sorry. I just took a look at the fbmessenger repo (https://github.com/rehabstudio/fbmessenger/). Is there a reason you're not just using the `QuickReply`  class (and the other relevant classes for what you're doing) and then using the `send()` function like in the `send_text_message` function here? https://github.com/RasaHQ/rasa_core/pull/456/files#diff-0a656bd3fec6f877069af1972c2b13bbR145

**Comment by advl03** [comment]
When I coded this part some time ago, I preferred to send the raw JSON: that approache gives me independency if Facebook adds or removes some fields of the message. That way, I didn't need to wait for a rehabstudio/fbmessenger release.

I agree that the proper way to send these messages would be to use the functions already included in the fbmessenger library. Unfortunately, I have no time now for a refactoring :-(

**Comment by tmbo** [comment]
I'll close this for the time being - but we definitely need to put some time into rethinking the way the output channels are implemented. It would be great to support more features of the custom channels.

**Pull request closed without merging**